btrack/_localization.py

```diff
@@ -44,23 +44,17 @@ def _centroids_from_single_arr(
     intensity_image: Optional[np.ndarray] = None,
     scale: Optional[Tuple[float]] = None,
     use_weighted_centroid: bool = False,
+    assign_class_ID: bool = False,
 ) -> np.ndarray:
     """Return the object centroids from a numpy array representing the
     image data."""
 
     if np.sum(segmentation) == 0:
         return {}
 
-    def _is_binary(x: np.ndarray) -> bool:
-        return ((x == 0) | (x == 1)).all()
-
-    def _is_ternary(x: np.ndarray) -> bool:
-        return ((x == 0) | (x == 1) | (x == 2)).all()
-
-    def _is_multiclass(x: np.ndarray) -> bool:
-        # check if image is not uniquely labelled
-        # test to see number unique labels is not the same as the max label
-        return np.max(label(x)) != np.max(x)
+    def _is_unique(x: np.ndarray) -> bool:
```

Comment:
oh also i tried to make the   https://github.com/quantumjot/BayesianTracker/blob/0be7225f958b75540ec6957622c9dd6b96f809bb/btrack/_localization.py#L47 a bit more versatile for ternary images and beyond (quaternary, ... etc) hence this change here.